### PR TITLE
Simplify TPUSpawn rank management

### LINKED ISSUE #10986: Property `TPUSpawnPlugin.tpu_global_core_rank` may be replaced by global rank property
## Proposed refactor

Deprecate `tpu_global_core_rank` in favor of `global_rank`.

### Pitch

It "looks" like tpu_global_core_rank in the TPUPlugin is just another name for what is known as the global rank. After #10896 we should investigate whether the two are equivalent and revert back to using the unified global_rank property exclusively. 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @kaushikb11 @rohitgr7 @justusschock @awaelchli @akihironitta @tchaton

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #10986 

The attributes `tpu_local_core_rank` and `tpu_global_core_rank` today behave identically to the `local_rank` and `global_rank` attributes. They are no longer needed.

It is a breaking change for everyone who was accessing them through the Strategy.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @kaushikb11 @rohitgr7 @justusschock @awaelchli @akihironitta @borda

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 1]
```diff
@@ -161,11 +147,17 @@ def process_dataloader(self, dataloader: DataLoader) -> MpDeviceLoader:
...
-        pass
+        # self._local_rank = process_idx
+        #
+        # if self.cluster_environment is None:
+        #     return
+        # self.cluster_environment.set_global_rank(self.node_rank * self.num_processes + self.local_rank)
+        # self.cluster_environment.set_world_size(self.num_nodes * self.num_processes)
+        # self.cluster_environment.set_global_rank(xm.get_ordinal())
+        # rank_zero_only.rank = self.cluster_environment.global_rank()
+        super().set_world_ranks(xm.get_local_ordinal())
```
Comment (kaushikb11) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:159]: @awaelchli Could I be of any help here?

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:159]: @kaushikb11 Yes that would be great! I'm wondering if `process_idx == xm.get_local_ordinal()` is true or if I'm missing something about these definitions here.

Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:159]: Yup, it should be true

File: CHANGELOG.md [hunk 2]
```diff
@@ -362,6 +362,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Removed `Strategy.optimizer_zero_grad` ([#11246](https://github.com/PyTorchLightning/pytorch-lightning/pull/11246))
 
+
+- Removed `TPUSpawnStrategy.{tpu_local_core_rank,tpu_global_core_rank}` c `TPUSpawnStrategy.{local_rank,global_rank}` ([#11163](https://github.com/PyTorchLightning/pytorch-lightning/pull/11163))
+
+
+- Removed `SingleTPUStrategy.{tpu_local_core_rank,tpu_global_core_rank}` attributes in favor of `SingleTPUStrategy.{local_rank,global_rank}`([#11163](https://github.com/PyTorchLightning/pytorch-lightning/pull/11163))
```
Comment (kaushikb11) [inline CHANGELOG.md:369]: `SingleTPUStrategy` wouldn't need `global_rank` ideally. `local_rank` should suffice.

**Review by kaushikb11 — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (awaelchli) [reply CHANGELOG.md:369]: that's true yes, it's just a special case. Should I remove mentioning it in the changelog?

**Review by justusschock — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (kaushikb11) [reply CHANGELOG.md:369]: Would be better to do a follow up PR to remove the property.

File: pytorch_lightning/strategies/tpu_spawn.py [hunk 3]
```diff
@@ -68,22 +68,8 @@ def __init__(
...
-    @property
-    def global_rank(self) -> int:
-        return self.tpu_global_core_rank
-
-    @property
-    def local_rank(self) -> int:
-        return self.tpu_local_core_rank
-
-    @property
-    def world_size(self) -> int:
```
Comment (tchaton) [inline pytorch_lightning/strategies/tpu_spawn.py:84]: Why do you remove world_size ?

File: pytorch_lightning/strategies/tpu_spawn.py [hunk 4]
```diff
@@ -267,8 +247,7 @@ def _wrapped_function(
 
     def _worker_setup(self, process_idx: int):
         reset_seed()
-        self.tpu_local_core_rank = xm.get_local_ordinal()
-        self.tpu_global_core_rank = xm.get_ordinal()
+        self.set_world_ranks(process_idx)
```
Comment (tchaton) [inline pytorch_lightning/strategies/tpu_spawn.py:250]: Should we introduce a TPUClusterEnv ?

File: pytorch_lightning/strategies/tpu_spawn.py (hunk 3 above)
Comment (awaelchli) [reply pytorch_lightning/strategies/tpu_spawn.py:84]: Because world size is set via `set_world_ranks` and then handled by LightningClusterEnvironment. I didn't remove it completely, it's just handled by the super class

File: pytorch_lightning/strategies/tpu_spawn.py (hunk 4 above)
Comment (awaelchli) [reply pytorch_lightning/strategies/tpu_spawn.py:250]: If there is a different way of launching tpu jobs than currently, then probably yes. If not, I don't see what it would do differently than the LightningEnvironment.

File: CHANGELOG.md (hunk 2 above)
Comment (awaelchli) [reply CHANGELOG.md:369]: Remove why? I think what you mean is to define it as constant 0.

**Review by akihironitta — COMMENTED** [review]
> It is a breaking change for everyone who was accessing them through the Strategy.

@awaelchli Are we removing these attributes without any deprecation? To me, it looks quite inconsistent that these are removed instantly in v1.6 while `SingleTPUPlugin` will be removed in v1.8.

**Comment by awaelchli** [comment]
> @awaelchli Are we removing these attributes without any deprecation?

Yes

> To me, it looks quite inconsistent that these are removed instantly in v1.6 while SingleTPUPlugin will be removed in v1.8.

Yes it is inconsistent because not all changes we want to do for 1.6 can be made backward compatible or deprecation-friendly.

I will add the deprecation if reviewers specifically request it.

File: pytorch_lightning/strategies/tpu_spawn.py (hunk 3 above)
Comment (tchaton) [reply pytorch_lightning/strategies/tpu_spawn.py:84]: Yes, but are you sure `self.node_rank * self.num_processes` are the right values ?

File: pytorch_lightning/strategies/tpu_spawn.py (hunk 4 above)
Comment (tchaton) [reply pytorch_lightning/strategies/tpu_spawn.py:250]: The ENV variable names used by TPU are different, so I would expect the code to be cleaner if fully encapsulated under a TPUClusterEnvironement.

**Comment by edward-io** [comment]
@awaelchli  is this PR ready for review?

**Comment by awaelchli** [comment]
@edward-io theoretically yes but merging is blocked by #11321 as it was requested by the reviewer. #11321 is failing the tpu tests which we need to debug first.

**Review by akihironitta — COMMENTED** [review]
LGTM!
